library/pk_wrap.c

```diff
@@ -735,22 +736,20 @@ static int ecdsa_verify_wrap(void *ctx_arg, mbedtls_md_type_t md_alg,
     unsigned char *p;
     psa_algorithm_t psa_sig_md = PSA_ALG_ECDSA_ANY;
     size_t curve_bits;
-    psa_ecc_family_t curve =
-        mbedtls_ecc_group_to_psa(ctx->grp.id, &curve_bits);
-    const size_t signature_part_size = (ctx->grp.nbits + 7) / 8;
+    psa_ecc_family_t curve = 0;
+    size_t signature_part_size;
     ((void) md_alg);
 
-    if (curve == 0) {
+    if (mbedtls_pk_get_ec_public_key_props(pk, &curve, &curve_bits) != 0) {
         return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
     }
+    signature_part_size = PSA_BITS_TO_BYTES(curve_bits);
 
     psa_set_key_type(&attributes, PSA_KEY_TYPE_ECC_PUBLIC_KEY(curve));
     psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_VERIFY_HASH);
     psa_set_key_algorithm(&attributes, psa_sig_md);
 
-    ret = mbedtls_ecp_point_write_binary(&ctx->grp, &ctx->Q,
-                                         MBEDTLS_ECP_PF_UNCOMPRESSED,
-                                         &key_len, buf, sizeof(buf));
+    ret = mbedtls_pk_get_public_key(pk, buf, sizeof(buf), &key_len);
```

Comment:
Right, this is totally useless (also the `mbedtls_pk_get_public_key` function itself since it used only here). I think I got slightly confused during the development of this PR by "who" is accessing internal data of the `pk_context` structure. I mean, if it's from "inside" the library, then there is no need for an extra function to get that kind of data: we can simply read the raw buffer directly.

Reply:
Clearly if it's from within PK itself, we can and should read the raw buffer directly. If it's from higher-level modules (X.509, TLS), then accessing struct members directly is technically possible, but we should think whether it's the right thing to do or if an internal API would be better - loose coupling and all that.

Probably depends on the situation, but in the past we've erred too much on the side of high-level modules relying on internal details of very low-level modules such as ECP or Bignum, so please do not blindly follow what existing code is doing when it comes to that kind of question :)